src/Form/Search.php

```diff
@@ -0,0 +1,157 @@
+        $collections = array_filter(array_combine($parents, 'islandora_object_load', $parents));
+      }
+      $show_label = $this->config('islandora_collection_search.settings')->get('islandora_collection_search_display_label');
+      if (!empty($collections)) {
+        // If there is more than one possible parents we can't differentiate for
+        // the "This collection" case, resort to using the labels.
+        if (count($collections) > 1) {
+          foreach ($collections as $object) {
+            $options[$object->id] = $object->label;
+          }
+        }
+        else {
+          $object = reset($collections);
+          $options[$object->id] = $show_label ? $object->label : $this->t('This collection');
+        }
+        $first_object = reset($collections);
+        $default_search = $first_object->id;
+      }
+      else {
+        $collection_object = islandora_object_load($collection);
+        $options[$collection_object->id] = $show_label ? $collection_object->label : $this->t('This collection');
+        $default_search = $collection_object->id;
+      }
+    }
+    $options = islandora_collection_search_retrieve_searchable_collections($options, $collection);
+    $all_pages = $this->config('islandora_collection_search.settings')->get('islandora_collection_search_all_pages');
+    if ($all_pages) {
+      if (!$collection) {
+        $collection = $all_pages;
+      }
+    }
+    $default_search_value = '';
+    // Check if we're on a search results page.
+    if ($this->config('islandora_collection_search.settings')->get('islandora_collection_search_retain_search_values') && strpos(Url::fromRoute('<current>')->toString(), 'islandora/search') === 0) {
+      $request_params = $this->getRequest()->query->all();
+      if (isset($request_params['cp'])) {
+        $default_search = $request_params['cp'];
+      }
+      $default_search_value = arg(2);
+    }
+    $form['simple']['collection_select'] = [
+      '#access' => $collection !== FALSE,
+      '#type' => 'select',
+      '#title' => $this->t('Select Collection'),
+      '#title_display' => 'invisible',
+      '#options' => $options,
+      '#default_value' => $default_search,
+    ];
+    $form['simple']["islandora_simple_search_query"] = [
+      '#size' => '15',
+      '#type' => 'textfield',
+      '#title' => $this->t('Search box'),
+      '#title_display' => 'invisible',
+      '#default_value' => $default_search_value,
+    ];
+    $form['simple']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('search'),
+    ];
+    $cache->applyTo($form);
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    $search_string = $form_state->getValue('islandora_simple_search_query');
+    // Replace the slash so URL doesn't break.
+    $search_string = islandora_solr_replace_slashes($search_string);
```

Comment:
Paranoia perhaps, but... this is in `islandora_solr/includes/utilities.inc`... we don't appear to have a load for it in here, anywhere, [while we did, previously](https://github.com/discoverygarden/islandora_collection_search/pull/23/files#diff-da7ed641f7e362bfd3c0b948ee67f0dfL126).